#!/bin/bash

#SBATCH -D ./ 
#SBATCH --job-name="join"
#SBATCH -o log-join.out
#SBATCH -c 1
#SBATCH -p ibismini
#SBATCH --mail-type=ALL
#SBATCH --mail-user=type_your_mail@ulaval.ca
#SBATCH --time=0-20:00
#SBATCH --mem=50000

cd $SLURM_SUBMIT_DIR

#usage: java JoinSingles chromosome_map_file [options] data=file1 [file2 file3...]

#variable
DIR="/home/jelel8/Software/LepMap2/v2016-04-27/bin/"
sc_lod=11
sc_sl=6

#define input
for i in $(ls 02_data/*linkage|sed 's/.linkage//g')
do
base=$(basename $i)

map="03_output/map_batch_1.genotypes_46.loc_lod"$sc_lod"_sizeLimit"$sc_sl".txt"     	#Initial LG map file. Typically generated by SeparateChromosomes or JoinSingles.
file="data=03_output/"$base"_trimmed_f.linkage"		#file [file2 ...]   Loads input genotypes in LINKAGE Pre-makeped format
lod="lodLimit=3"				# LOD score limit [10.0]
#ld="lodDifference=NUM"		       		#Required difference in LOD scores between the best LG and the second best LG of each joined marker[0.0]
#up="(fe)malePrior=NUM"		       		#Uniform prior for recombination fraction for (maternal) paternal haplotypes [maternal:2*epsilon*(1-epsilon)]
#ap="(fe)malePrior=A B C"	     		#Affine prior for recombination fraction for (maternal) paternal haplotypes [paternal:4 0 0.5]
#t="(fe)maleTheta=NUM"		       		#Use fixed recombination fraction [not set]
#im="informativeMask=23"	     			#Use only markers with informative father (1), mother(2), both parents(3) or neither parent(0) [0123]
#f="families=f1 [f2 ...]"	    		#Use only families f1 f2 ... [not set]

java -cp $DIR JoinSingles $map $lod $ld $up $ap $t $im $f $file >03_output/map_"$base"_js.txt

done 2>&1 | tee 98_log_files/"$TIMESTAMP"_joinsingles.log
