#!/bin/bash
#$ -N log.lepmap2_JoinSingle
#$ -M userID@ulaval.ca
#$ -m beas
#$ -pe smp 1
#$ -l h_vmem=10G
#$ -l h_rt=03:00:00
#$ -cwd
#$ -S /bin/bash

#usage: java JoinSingles chromosome_map_file [options] data=file1 [file2 file3...]

DIR="/home/jelel8/Software/LepMap2/bin/"
map="03-output/map.lod10.sizeLimit1.txt"     	#Initial LG map file. Typically generated by SeparateChromosomes or JoinSingles.
file="03-output/map_trimmed_f.linkage"		#file [file2 ...]   Loads input genotypes in LINKAGE Pre-makeped format
lod="lodLimit=4"				# LOD score limit [10.0]
#ld="lodDifference=NUM"		       		#Required difference in LOD scores between the best LG and the second best LG of each joined marker[0.0]
#up="(fe)malePrior=NUM"		       		#Uniform prior for recombination fraction for (maternal) paternal haplotypes [maternal:2*epsilon*(1-epsilon)]
#ap="(fe)malePrior=A B C"	     		#Affine prior for recombination fraction for (maternal) paternal haplotypes [paternal:4 0 0.5]
#t="(fe)maleTheta=NUM"		       		#Use fixed recombination fraction [not set]
im="informativeMask=23"	     			#Use only markers with informative father (1), mother(2), both parents(3) or neither parent(0) [0123]
#f="families=f1 [f2 ...]"	    		#Use only families f1 f2 ... [not set]

java -cp $DIR JoinSingles $map $lod $ld $up $ap $t $im $f $file >03-output/map_js.txt
