#!/bin/bash
#$ -N log.lepmap2_JoinSingle
#$ -M userID@ulaval.ca
#$ -m beas
#$ -pe smp 1
#$ -l h_vmem=20G
#$ -l h_rt=00:20:00
#$ -cwd
#$ -S /bin/bash

#EstimateLod deprecated
#java EstimateLODLimit data=/data4/sandrine/11_Lep_Map/data/Uneak_210_80_75_premakeped_file_20151002.linkage malePrior=-1

INPUT="/home/jelel8/file_BEN_final_f"
DIR="/home/jelel8/Software/LepMap2/bin/"
MAP="map.lod22.NoinfoMask24.sizeLimit22.txt"
infoMask=2

java -cp $DIR JoinSingles $MAP malePrior=0.01 data="$INPUT".linkage lodLimit="$LOD" >map.Lod22.JsLod"$LOD"_js.sizeLimit22.txt

: ' 
usage: java JoinSingles chromosome_map_file [options] data=file1 [file2 file3...]
options:
         chromosome_map_file     Initial LG map file. Typically generated by SeparateChromosomes or JoinSingles.
         data=file [file2 ...]   Loads input genotypes in LINKAGE Pre-makeped format
         lodLimit=NUM [NUM2 ...] LOD score limit [10.0]
         lodDifference=NUM       Required difference in LOD scores between the best LG and the second best LG of each joined marker[0.0]
         (fe)malePrior=NUM       Uniform prior for recombination fraction for (maternal) paternal haplotypes [maternal:2*epsilon*(1-epsilon)]
         (fe)malePrior=A B C     Affine prior for recombination fraction for (maternal) paternal haplotypes [paternal:4 0 0.5]
         (fe)maleTheta=NUM       Use fixed recombination fraction [not set]
         informativeMask=STR     Use only markers with informative father (1), mother(2), both parents(3) or neither parent(0) [0123]
         families=f1 [f2 ...]    Use only families f1 f2 ... [not set]
'
